### ISSUE TITLE
Select: onSelected should be onSelect

### ISSUE THREAD

**Issue by willdurand** [Zoapp/react-materialcomponents#52]
We should rename the `onSelected` prop to `onSelect` to be consistent with the other props in the other components (e.g., `onChange`).

In addition, it would be nice to specify a `value` that is the current/selected value.

**Comment by mikbry** [comment]
selectedIndex : is the prop to specify the current selected item
Yes on for onSelect

**Comment by willdurand** [comment]
> selectedIndex : is the prop to specify the current selected item

yes but that's usually not what we store. we want the (selected) `value` usually, not the index and it would be useful to pass this value to the `select`.

**Comment by mikbry** [comment]
I conform to MDC specs:
https://material.io/components/web/catalog/input-controls/select-menus/

**Comment by willdurand** [comment]
I don't see where it is written to not use `value` in the spec 😉

Note: all react components use `onChange/value` to control form components.

**Comment by mikbry** [comment]
Ok let's follow React. I also used onChange in other components.

**Comment by willdurand** [comment]
Ok so I fixed various things in the `Select` component, I did not rename the prop yet though, but the rest is ok, when you pass a `selectedIndex` it displays the selected item as it should and it returns both the selected value and index.

Selected value is the `value` prop on the item component if specified, otherwise it is `children`.
